pkg/pdp/types/zap.go

```diff
@@ -0,0 +1,97 @@
+package types
+
+import (
+	"github.com/google/uuid"
+	"go.uber.org/zap/zapcore"
+)
+
+// MarshalLogObject implements zapcore.ObjectMarshaler for ProofSetStatus
+func (p ProofSetStatus) MarshalLogObject(enc zapcore.ObjectEncoder) error {
```

Comment:
again nice refactor! again for the future I would leave this sort of thing out of an already large PR. but nb